akka-javasdk/src/main/scala/akka/javasdk/impl/agent/AgentEffectImpl.scala

```diff
@@ -165,7 +165,8 @@ private[javasdk] final class BaseAgentEffectBuilder[Reply]
   }
 
   override def map[T](mapper: function.Function[String, T]): MappingResponseBuilder[T] = {
-    updateRequestModel(_.copy(responseType = classOf[String]))
+    updateRequestModel(
+      _.copy(responseType = classOf[String], responseMapping = Some(mapper.asScala.asInstanceOf[Function1[Any, Any]])))
```

Comment:
It's weird because the response type is string by default `responseType = classOf[String],` here https://github.com/akka/akka-sdk/blob/c227d26c952e924b580113985740ca9a831150e4/akka-javasdk/src/main/scala/akka/javasdk/impl/agent/AgentEffectImpl.scala#L39

So why do we need to set that again?

Reply:
removed